### PR TITLE
Internal: Class fix for better maintainability

### PR DESCRIPTION
**Pull request by ronenboxer**

### CODE REVIEW

File: modules/checklist/steps/step-base.php [hunk 1]
```diff
@@ -138,6 +166,22 @@ public function get_step_initial_progress() : array {
 		return $initial_progress;
 	}
 
+	/**
+	 * @return array
+	 */
+	public function get_config() : array {
```
Comment (TzviRabinovitch) [inline modules/checklist/steps/step-base.php:172]: SRP ?

Comment (ronenelementor) [reply modules/checklist/steps/step-base.php]: Moved to manager as discussed

File: modules/checklist/steps/step-base.php [hunk 2]
```diff
@@ -85,25 +111,12 @@ public function unmark_as_completed() : void {
 	 * @return void
 	 */
 	public function maybe_mark_as_completed() : void {
-		if ( $this->step_config[ self::IS_COMPLETION_IMMUTABLE ] && $this->is_absolute_completed() ) {
+		if ( $this->get_is_completion_immutable() && $this->is_absolute_completed() ) {
```
Comment (TzviRabinovitch) [inline modules/checklist/steps/step-base.php:114]: maybe check current status ?

File: modules/checklist/steps-manager.php [hunk 3]
```diff
@@ -90,6 +100,26 @@ public function get_step_by_id( string $step_id ) : ?Step_Base {
...
+		return $step_instance
+			? [
+				'id' => $step_instance->get_id(),
+				'title' => $step_instance->get_title(),
+				'description' => $step_instance->get_description(),
+				'learn_more_text' => $step_instance->get_learn_more_text(),
+				'learn_more_url' => $step_instance->get_learn_more_url(),
+				'cta_text' => $step_instance->get_cta_text(),
+				'cta_url' => $step_instance->get_cta_url(),
+				Step_Base::IS_COMPLETION_IMMUTABLE => $step_instance->get_is_completion_immutable(),
```
Comment (davseve) [inline modules/checklist/steps-manager.php:118]: Image Url?

**Review by davseve — DISMISSED** [review]

**Review by TzviRabinovitch — APPROVED** [review]

**Pull request merged**